### PR TITLE
refactor(webapp): Refactor timeline selection

### PR DESCRIPTION
**Pull request by pavelpashkovsky**
## Brief
- https://github.com/pyroscope-io/pyroscope/issues/1668

## Changes
- DEMO: https://pr-1734.pyroscope.io/
-  disable interraction with selection boundaries on timeline with `selectionType="double"`
-  refactor types for Selection.plugin
- remove custom Flot types

### CODE REVIEW

File: webapp/javascript/components/TimelineChart/Selection.plugin.ts [hunk 1]
```diff
@@ -1,16 +1,35 @@
-/* eslint-disable */
+/* eslint-disable eqeqeq */
```
Comment (eh-am) [inline webapp/javascript/components/TimelineChart/Selection.plugin.ts:1]: typo?

Comment (pavelpashkovsky) [reply webapp/javascript/components/TimelineChart/Selection.plugin.ts:1]: no, it's https://eslint.org/docs/latest/rules/eqeqeq

I didn't take a risk to transform all `==` to `===`

Comment (eh-am) [reply webapp/javascript/components/TimelineChart/Selection.plugin.ts:1]: ohh

can you add a message explaining that, bc i can see most people ignoring it haha
https://eslint.org/docs/latest/user-guide/configuring/rules#using-configuration-comments

Comment (pavelpashkovsky) [reply webapp/javascript/components/TimelineChart/Selection.plugin.ts]: updated

Comment (eh-am) [reply webapp/javascript/components/TimelineChart/Selection.plugin.ts]: Not sure I understood, i was asking for a comment explaining the use of `eqeqeq`. As I linked above, you use `--` to delimit the comments. Eg
```
/* eslint eqeqeq: "off", curly: "error" -- Here's a description about why this configuration is necessary. */
```

Comment (pavelpashkovsky) [reply webapp/javascript/components/TimelineChart/Selection.plugin.ts]: ah, really, please check it one more time 😅

**Review by eh-am — APPROVED** [review]

**Pull request merged**